infor-design/enterprise-wc
# 1045 - IdsDataGrid dropdown editor (M3)

**Explain the details for making this change. What existing problem does the pull request solve?**
Added grid support for dropdown editor. This PR only partially addresses the #1045 ticket, specifically (M3) 

NOTE: Dropdown editors near the bottom of grid will look cut off. Might be addressed [here](https://github.com/infor-design/enterprise-wc/issues/1065) or in another future ticket

**Related github/jira issue (required)**:
Partially Closes #1045 

**Steps necessary to review your pull request (required)**:
1. Checkout branch, `npm install`, `npm run start`
2. Go to http://localhost:4300/ids-data-grid/editable.html
3. Test Currency Column dropdown editor 
4. Test in different themes, languages

**Validation/Dirty**
- Change any currency cell value, dirty indicator should show when edit is finished
- Change any currency cell value to the empty value, it should show an error when edit is finished

**Keyboard**
- [Enter] Activate editing
- [ArrowDown] Opens Dropdown
- [ArrowDown/ArrowUp] Navigates Dropdown list
- [Enter] Selects value and starts editing the same cell in next row
- [Shift + Enter] Selects value and starts editing the same cell in previous row
- [Esc] Cancels editing
- [Tab] Finish editing and start editing the next cell
- [Shift + Tab] Finish editing and start editing the previous cell

**Included in this Pull Request**:
